code/modules/shuttle/shuttle.dm

```diff
@@ -453,6 +453,36 @@
 	else
 		WARNING("shuttle \"[name]\" could not enter transit space. S0=[S0 ? S0.name : "null"] S1=[S1 ? S1.name : "null"]")
 
+/**
+ * Scuttle the ship
+ *
+ * Delete all of the areas, and delete any cryopods
+ */
+/obj/docking_port/mobile/proc/mothball()
+	var/obj/docking_port/stationary/current_dock = get_docked()
+
+	var/underlying_area_type = SHUTTLE_DEFAULT_UNDERLYING_AREA
+	if(current_dock && current_dock.area_type)
+		underlying_area_type = current_dock.area_type
+
+	var/list/old_turfs = return_ordered_turfs(x, y, z, dir)
+
+	var/area/underlying_area = GLOB.areas_by_type[underlying_area_type]
+	if(!underlying_area)
+		underlying_area = new underlying_area_type(null)
+
+	for(var/i in 1 to old_turfs.len)
+		var/turf/oldT = old_turfs[i]
+		if(!oldT || !istype(oldT.loc, area_type))
+			continue
+		var/obj/machinery/cryopod/pod = locate() in oldT.contents
```

Comment:
This is probably poopy and should be some sort of callback, but im not quite sure where i would put it, if anyone has any ideas lmk